sway-core/src/optimize.rs

```diff
@@ -2148,6 +2148,124 @@ impl FnCompiler {
                             }
                         }
                     }
+                    Type::Union(_) | Type::String(_) => {
+                        // Use state_load_quad_word/state_store_quad_word as many times as needed
+                        // using sequential keys
+
+                        // First, create a name for the value to load from or store to
+                        let mut value_name = format!("{}{}", "val_for_", ix.to_usize());
+                        for ix in &indices {
+                            value_name = format!("{}_{}", value_name, ix);
+                        }
+                        let alias_value_name =
+                            self.lexical_map.insert(value_name.as_str().to_owned());
```

Comment:
This ends up being more lines, even though it's a 'one-liner'.  Well, at least the obsolete `.as_str().to_owned()` should go.
```suggestion
                        let value_name = format!(
                            "val_for_{}{}",
                            ix.to_usize(),
                            indices
                                .iter()
                                .map(|idx| format!("_{idx}"))
                                .collect::<Vec<_>>()
                                .join("")
                        );
                        let alias_value_name = self.lexical_map.insert(value_name);
```